src/sentry/rules/processing/delayed_processing.py

```diff
@@ -463,6 +464,39 @@ def bucket_num_groups(num_groups: int) -> str:
     return "1"
 
 
+CHUNK_PAGE_SIZE = 10000
+
+
+def process_rulegroups_in_batches(project_id: int, batch_size=CHUNK_PAGE_SIZE):
+    event_count = buffer.backend.get_hash_length(Project, {"project_id": project_id})
+
+    if event_count < CHUNK_PAGE_SIZE:
+        return apply_delayed(project_id)
+
+    # if the dictionary is large, get the items and chunk them.
+    rulegroup_to_event_data = fetch_rulegroup_to_event_data(project_id)
+
+    with metrics.timer("delayed_processing.process_batch.duration"):
+        items = iter(rulegroup_to_event_data.items())
+        while True:
+            batch = dict(islice(items, batch_size))
+            if not batch:
+                break
+
+            batch_key = f"{project_id}:{uuid.uuid4()}"
+            batch_values = json.dumps(batch)
```

Comment:
this is likely to not work.. more likely i'll need to iterate each item in the batch, adding each value to the new redis store, and removing it from the old. :sob: